### PR TITLE
ci: added github actions

### PR DESCRIPTION
**Pull request by MoumitaM**
**Description**
Introduced GitHub actions.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,8 +4,22 @@
...
+    "check:circular": "madge --circular --extensions js ./src || exit 0",
+    "check:duplicates": "jscpd src --threshold 5",
+    "check:security": "npm audit --recursive --audit-level=high",
+    "format": "prettier --write .",
+    "lint:fix": "eslint . --fix",
+    "prepare": "husky install",
+    "pre-commit": "npm run test && npx lint-staged",
+    "commit-msg": "commitlint --edit",
+    "commit": "git-cz",
+    "test": "exit 0",
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply package.json:21]: Addressed

File: .github/workflows/draft-new-release.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      - name: Update changelog & bump version
+        id: finish-release
+        env:
+          HUSKY: 0
+        run: |
+          echo "Current version: $CURRENT_VERSION_VALUE"
+          echo "New version: $NEW_VERSION_VALUE"
+          npx replace $CURRENT_VERSION_VALUE $NEW_VERSION_VALUE sonar-project.properties
+          git add sonar-project.properties
+          npx standard-version -a
```
Comment (bardisg) [inline .github/workflows/draft-new-release.yml:72]: npx standard-version -a --skip.tag

File: .github/workflows/draft-new-release.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+      - name: Push new version in release branch & tag
+        run: |
+          git push --follow-tags
```
Comment (bardisg) [inline .github/workflows/draft-new-release.yml:76]: git push

File: .github/workflows/publish-new-release.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+
+      - name: Create GitHub Release
+        id: create_release
+        env:
+          HUSKY: 0
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CONVENTIONAL_GITHUB_RELEASER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
```
Comment (bardisg) [inline .github/workflows/publish-new-release.yml:42]: git tag -a v${{ steps.extract-version.outputs.release_version }} -m "chore: release v${{ steps.extract-version.outputs.release_version }}"
git push origin refs/tags/v${{ steps.extract-version.outputs.release_version }}
npx conventional-github-releaser -p angular

File: .github/workflows/publish-new-release.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          npx conventional-github-releaser -p angular
+
+      - name: Create pull request into staging
+        uses: repo-sync/pull-request@v2
+        with:
+          source_branch: 'main'
+          destination_branch: 'deevelop'
```
Comment (bardisg) [inline .github/workflows/publish-new-release.yml:49]: develop (typop has two ee)

File: sonar-project.properties [hunk 6]
```diff
@@ -0,0 +1,35 @@
+sonar.log.level=INFO
+sonar.verbose=false
+sonar.qualitygate.wait=false
+
+# Project details
+sonar.projectKey=rudderlabs_gatsby-plugin-rudderstack
+sonar.organization=rudderlabs
+sonar.projectName=gatsby-plugin-rudderstack
+sonar.projectVersion=2.21.0
```
Comment (bardisg) [inline sonar-project.properties:9]: 2.0.0 (should match the package json version)

**Comment by MoumitaM** [comment]
Included all the comments

**Review by bardisg — APPROVED** [review]

**Pull request merged**